ietf/doc/views_statement.py

```diff
@@ -17,4 +29,247 @@ def serve_pdf(self, name, rev=None):
     if not p.exists():
         raise Http404
     else:
-        return FileResponse(p.open(mode="rb"),content_type="application/pdf")
+        return FileResponse(p.open(mode="rb"), content_type="application/pdf")
+
+
+class StatementUploadForm(forms.Form):
+    ACTIONS = [
+        ("enter", "Enter content directly"),
+        ("upload", "Upload content from file"),
+    ]
+    statement_submission = forms.ChoiceField(choices=ACTIONS, widget=forms.RadioSelect)
+    statement_file = forms.FileField(
+        label="Markdown or PDF source file to upload", required=False
+    )
+    statement_content = forms.CharField(
+        widget=forms.Textarea(attrs={"rows": 30}), required=False, strip=False
+    )
+
+    def clean(self):
+        def require_field(f):
+            if not self.cleaned_data.get(f):
+                self.add_error(f, forms.ValidationError("You must fill in this field."))
+                return False
+            else:
+                return True
+
+        submission_method = self.cleaned_data.get("statement_submission")
+        markdown_content = ""
+        if submission_method == "enter":
+            if require_field("statement_content"):
+                markdown_content = self.cleaned_data["statement_content"].replace(
+                    "\r", ""
+                )
+                default_content = render_to_string(
+                    "doc/statement/statement_template.md", {"settings": settings}
+                )
+                if markdown_content == default_content:
+                    raise forms.ValidationError(
+                        "The example content may not be saved. Edit it to contain the next revision statement content."
+                    )
+                if (
+                    markdown_content
+                    == "The current revision of this statement is in pdf format"
```

Comment:
Duplicating this magic string is fragile, might want to make it a const so someone doesn't break this protection when they rephrase the message.